src/devtools/tests/devtools.test.tsx

```diff
@@ -0,0 +1,321 @@
+import React from 'react'
+import {
+  fireEvent,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from '@testing-library/react'
+import { QueryClient, QueryCache, useQuery } from '../..'
+import { getByTextContent, renderWithClient, sleep } from './utils'
+
+describe('ReactQueryDevtools', () => {
+  const queryCache = new QueryCache()
+  const queryClient = new QueryClient({
+    queryCache,
+    defaultOptions: {
+      queries: {
+        staleTime: 0,
+      },
+    },
+  })
+
+  beforeEach(() => {
+    queryCache.clear()
+  })
+
+  it('should be able to open and close devtools', async () => {
+    function Page() {
+      const { data = 'default' } = useQuery('check', async () => {
+        await sleep(10)
+        return 'test'
+      })
+
+      return (
+        <div>
+          <h1>{data}</h1>
+        </div>
+      )
+    }
+
+    renderWithClient(queryClient, <Page />, { initialIsOpen: false })
+
+    // Since the initial is open state is false, expect the close button to not be present
+    // in the DOM. Then find the open button and click on it.
+    const closeButton = screen.queryByRole('button', {
+      name: /close react query devtools/i,
+    })
+    expect(closeButton).toBeNull()
+    fireEvent.click(
+      screen.getByRole('button', { name: /open react query devtools/i })
+    )
+
+    // Wait for the animation to finish and the open button to be removed from DOM once the devtools
+    // is opened. Then find the close button and click on it.
+    await waitForElementToBeRemoved(() =>
+      screen.queryByRole('button', { name: /open react query devtools/i })
+    )
+    fireEvent.click(
+      screen.getByRole('button', { name: /close react query devtools/i })
+    )
+
+    // Finally once the close animation is completed expect the open button to
+    // be present in the DOM again.
+    await screen.findByRole('button', { name: /open react query devtools/i })
+  })
+
+  it('should display the correct query states', async () => {
+    function Page() {
+      const { data = 'default' } = useQuery(
+        'check',
+        async () => {
+          await sleep(100)
+          return 'test'
+        },
+        { staleTime: 300 }
+      )
+
+      return (
+        <div>
+          <h1>{data}</h1>
+        </div>
+      )
+    }
+
+    function PageParent() {
+      const [isPageVisible, setIsPageVisible] = React.useState(true)
+
+      return (
+        <div>
+          <button
+            type="button"
+            aria-label="Toggle page visibility"
+            onClick={() => setIsPageVisible(visible => !visible)}
+          >
+            Toggle Page
+          </button>
+          {isPageVisible && <Page />}
+        </div>
+      )
+    }
+
+    renderWithClient(queryClient, <PageParent />)
+
+    fireEvent.click(
+      screen.getByRole('button', { name: /open react query devtools/i })
+    )
+
+    // Find the current query from the cache
+    const currentQuery = queryCache.find('check')
```

Comment:
let's not over-extend on comments please. The comment here (`find the query from the cache`) doesn't add much which the code itself doesn't tell us via: `queryCache.find`. It does however make the flow of the code harder to read if every line of code has 3 lines of comments stating the WHAT is happening.

So let's try to keep comments describing maybe the WHY something is happening, if we need them at all. tests written with react-testing-library are pretty declarative and self-explanatory, so only a few comments should be needed please.

Reply:
Thanks for pointing that out 🙌  . Toned down the unnecessary comments and kept only the ones that I think are necessary to understand why something is happening